### PR TITLE
Fix: disable fields when extenstion is not installed in create decision action

### PR DESCRIPTION
**Pull request by krawczykkatarzyna**
https://tw.auroracreation.com/app/tasks/15687539

- disable fields when extension is not installed
- display banner
- redirect to extension

Screen Shot 2023-12-04 at 14 16 53

### CODE REVIEW

**Review by melyndav — DISMISSED** [review]
Sorry, I left a comment on the wrong PR. Please ignore.

**Review by arrenv — CHANGES_REQUESTED** [review]
Thank you @krawczykkatarzyna!

I noticed these things while reviewing.

1. This should only work if the `VotingReputation` extension is been `Enabled`. At the moment, it is only checking if the extension has been installed. 

2. The 'Permission' Decision method is not possible at all with this Action type at the moment, so, the dropdown should only support 'Reputation'.
image


4. I noticed this error in the console when creating an Agreement.
image

5. After creating an agreement with the Reputation decision method I don't see the description of the Agreement or the title.
image

6. Likely out of scope of this PR, but I also noticed that the Agreement dashboard widget did not update after creating an agreement.
image

File: src/components/v5/common/ActionSidebar/consts.tsx [hunk 1]
```diff
@@ -74,3 +53,49 @@ export const ACTION_BASE_VALIDATION_SCHEMA = object()
       ),
   })
   .defined();
+
+export const useCreateActionTypeNotification = (
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/consts.tsx:57]: can this be a component? it's a hook returning JSX inside a file that should containt constants
Seeing how it's used, we can just replace this codesnip
image
with a component called `ActionTypeNotification` which can then just render the banner conditionally

File: src/components/v5/common/ActionSidebar/partials/ActionButtons.tsx [hunk 2]
```diff
@@ -5,7 +5,11 @@ import { useMobile } from '~hooks';
 import Button, { TxButton } from '~v5/shared/Button';
 import { useActionSidebarContext } from '~context/ActionSidebarContext';
 import { ActionButtonsProps } from '../types';
-import { useSubmitButtonDisabled, useSubmitButtonText } from './hooks';
+import {
+  useDisableFields,
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/ActionButtons.tsx:9]: hope it's not too nitpicky, but `useDisableFields` makes it sound like it's going to disable fields, maybe rename to `useIsFieldDisabled`

File: src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/partials/SidebarBanner.tsx [hunk 3]
```diff
@@ -1,15 +1,16 @@
 import React, { FC } from 'react';
 
 import { useFormContext, useWatch } from 'react-hook-form';
+import { noop } from 'lodash';
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/partials/SidebarBanner.tsx:4]: you can use noop from our utils

File: src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/partials/SidebarBanner.tsx [hunk 4]
```diff
@@ -21,28 +22,54 @@ export const SidebarBanner: FC = () => {
...
           icon="warning-circle"
           callToAction={
-            <a
-              href={actionTypeNofiticationHref}
-              target="_blank"
-              rel="noreferrer"
-            >
-              {formatText({ id: 'learn.more' })}
-            </a>
+            <>
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/partials/SidebarBanner.tsx:55]: probably no need for a fragment here

File: src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/partials/SidebarBanner.tsx [hunk 5]
```diff
@@ -21,28 +22,54 @@ export const SidebarBanner: FC = () => {
...
   );
 
-  const actionTypeNotificationTitle =
+  const actionTypeNotification =
     useCreateActionTypeNotification(selectedAction);
   const actionTypeNofiticationHref =
     useCreateActionTypeNotificationHref(selectedAction);
 
-  if (actionTypeNotificationTitle) {
+  const getProperty = (obj, key, defaultValue) => {
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/ActionSidebarContent/partials/SidebarBanner.tsx:30]: this looks like a useful util! maybe we can already put it to `utils/objects` already :)

**Review by bassgeta — CHANGES_REQUESTED** [review]
Good work! Dropped just a few minor comments

File: src/components/v5/shared/ActionTypeNotification/ActionTypeNotification.tsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+                      toggleActionSidebarOff();
+                    }}
+                  >
+                    {formatText(MSG.viewExtension)}
+                  </button>
+                )
+              )
+            }
+          >
+            {actionTypeNotificationTitle()}
```
Comment (bassgeta) [inline src/components/v5/shared/ActionTypeNotification/ActionTypeNotification.tsx:118]: we can reuse notificationTitle here

File: src/components/v5/common/ActionSidebar/partials/hooks.ts [hunk 7]
```diff
@@ -56,3 +58,20 @@ export const useSubmitButtonDisabled = () => {
 
   return false;
 };
+
+export const useIsFieldDisabled = () => {
+  const { extensionData } = useExtensionData(Extension.VotingReputation);
+  const selectedAction: Action | undefined = useWatch({
+    name: ACTION_TYPE_FIELD_NAME,
+  });
+  const isExtenstionEnabled =
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/hooks.ts:67]: isExtensionEnabled

**Review by bassgeta — CHANGES_REQUESTED** [review]

**Review by bassgeta — APPROVED** [review]

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
Nice work with this! The only gripe I could find is that the description is not aligned when it's not expanded on the summary side panel after submit:
FireShot Capture 265 - Transaction - 0xa592ab98a4a8fadc87acb2205aad6849000ce7dd893f0f8e17f4b_ - localhost

File: src/components/v5/shared/RichText/RichText.tsx [hunk 8]
```diff
@@ -119,7 +120,9 @@ const RichText: FC<RichTextProps> = ({
                 className={clsx('transition sm:hover:text-blue-400 text-left', {
                   'text-gray-900': characterCount,
                   'text-gray-400': !characterCount,
+                  'sm:hover:text-blue-400': !isDisabled,
```
Comment (iamsamgibbs) [inline src/components/v5/shared/RichText/RichText.tsx:123]:
```suggestion
                className={clsx('transition text-left', {
                  'text-gray-900': characterCount,
                  'text-gray-400': !characterCount,
                  'sm:hover:text-blue-400': !isDisabled,
```

sm:hover:text-blue-400 should no longer be applied by default as it is now applied only when !isDisabled.

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
This is looking great! Description alignment is fixed. I've only noticed one very minor thing on my end. The "enter a description" button still has a hover state when disabled. I've added a comment where this is occurring.

Screenshot 2024-01-24 at 13 08 50

**Review by iamsamgibbs — APPROVED** [review]
Looks good to me!

**Review by ArmandoGraterol — DISMISSED** [review]
Looking good to me 👍 Thank you for the changes

**Review by iamsamgibbs — DISMISSED** [review]
Looks good to me!

**Review by arrenv — APPROVED** [review]
Looks good to me, thank you @CzarekDryl.

Out of scope of this PR, put just noting that Creating an Agreement is unfortunately not working for me.

image

**Review by iamsamgibbs — APPROVED** [review]

File: src/components/v5/shared/ActionTypeNotification/ActionTypeNotification.tsx [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+  const {
+    actionSidebarToggle: [, { toggleOff: toggleActionSidebarOff }],
+  } = useActionSidebarContext();
+
+  const actionTypeNotificationHref =
+    (selectedAction === ACTION.UNLOCK_TOKEN &&
+      'https://docs.colony.io/use/managing-funds/unlock-token/') ||
+    undefined;
+
+  const actionTypeNotificationTitle = (): string | undefined => {
```
Comment (ArmandoGraterol) [inline src/components/v5/shared/ActionTypeNotification/ActionTypeNotification.tsx:63]: I think this name doesn't reflect that this is a function. Perhaps something like `getNotificationTitle` would be better

File: src/components/v5/shared/ActionTypeNotification/ActionTypeNotification.tsx [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+        break;
+      case ACTION.CREATE_DECISION:
+        title = isFieldDisabled
+          ? formatText(MSG.createDecisionError)
+          : undefined;
+        break;
+      default:
+        title = undefined;
+    }
+    return title;
```
Comment (ArmandoGraterol) [inline src/components/v5/shared/ActionTypeNotification/ActionTypeNotification.tsx:83]: We don't need `let title` here. We can directly return the title in each case:
```
  const actionTypeNotificationTitle = (): string | undefined => {
    switch (selectedAction) {
      case ACTION.UNLOCK_TOKEN:
        return formatText(
          isNativeTokenUnlocked ? MSG.unlockedToken : MSG.unlockedTokenError,
        );
      case ACTION.ENTER_RECOVERY_MODE:
        return formatText(MSG.enterRecoveryModeError);
      case ACTION.CREATE_DECISION:
        return isFieldDisabled
          ? formatText(MSG.createDecisionError)
          : undefined;
        break;
      default:
        return undefined;
    }
  };

  const notificationTitle = actionTypeNotificationTitle();
```

**Review by arrenv — DISMISSED** [review]
Still good!

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
It's working as expected as far as I can tell! Noticed some small things that we could change/improve but other than that, this should be good to go

**Review by ArmandoGraterol — DISMISSED** [review]
LGTM 🆗

**Review by iamsamgibbs — APPROVED** [review]
All looks good to me!

**Review by iamsamgibbs — APPROVED** [review]

**Review by KornelKwiatkowski — APPROVED** [review]

**Pull request merged**